apps/src/lib/levelbuilder/code-docs-editor/ProgrammingExpressionEditor.jsx

```diff
@@ -43,13 +47,22 @@ export default function ProgrammingExpressionEditor({
       return;
     }
     setIsSaving(true);
+    const copiedParameters = programmingExpression.parameters.map(p => {
+      const copied = {...p};
+      delete copied.key;
+      return copied;
+    });
+    const programmingExpressionToSave = {
+      ...programmingExpression,
+      parameters: JSON.stringify(copiedParameters)
```

Comment:
does this need to be stringified? we're stringifying the surrounding expression already

Reply:
It has to do with the way parameters are permitted in Rails but the [docs](https://www.rubydoc.info/docs/rails/ActionController%2FParameters:permit) have some secret syntax I can use!